pkg/scaling/scale_handler.go

```diff
@@ -681,3 +711,126 @@ func (h *scaleHandler) isScaledJobActive(ctx context.Context, scaledJob *kedav1a
 	logger.V(1).WithValues("ScaledJob", scaledJob.Name).Info("Checking if ScaleJob Scalers are active", "isActive", isActive, "maxValue", maxFloatValue, "MultipleScalersCalculation", scaledJob.Spec.ScalingStrategy.MultipleScalersCalculation)
 	return isActive, queueLength, maxValue
 }
+
+// getTrueMetricArray is a help function made for composite scaler to determine
+// what metrics should be used. In case of composite scaler (ComplexScalingLogic struct),
+// all external metrics will be used. Returns all external metrics otherwise it
+// returns the same metric given.
+func (h *scaleHandler) getTrueMetricArray(ctx context.Context, metricName string, so *kedav1alpha1.ScaledObject) ([]string, error) {
+	// if composite scaler is given return all external metrics
+	if so != nil && so.Spec.Advanced != nil && so.Spec.Advanced.ComplexScalingLogic.Target != "" {
+		if len(so.Status.ExternalMetricNames) == 0 {
+			scaledObject := &kedav1alpha1.ScaledObject{}
+			err := h.client.Get(ctx, types.NamespacedName{Name: so.Name, Namespace: so.Namespace}, scaledObject)
```

Comment:
Don't you already have the information in current `so` var? I mean, IIRC, the `so` var is recovered in the beginning of the loop, so it should contain all the info and it could save an api call

Reply:
this was made for a bug, where metrics are not cached because of a bug (I believe). We looked at this with @zroubalik he would know more. Im checking if the Metric names are 0 (if this bug occurred) and if so Im fetching the SO again. If the bug is fixed, this doesnt have to be there at all.

Reply:
are we tracking somewhere that bug @zroubalik ?

Reply:
I recall this problem, but my brain is empty TBH, so not sure if we track this 😄 🤦 

 @gauron99 could you please open an issue to track this problem?